capstone/src/main/webapp/visualizationdata.js

```diff
@@ -208,11 +208,134 @@ function getNumWeeksBetween(date1, date2) {
   return Math.round(Math.abs(date2 - date1) / MILLISECONDS_ONE_WEEK);
 }
 
+/**
+ * Returns data for the bar chart visualization.
+ * @return data object of the form:
+ * {
+ *     'data': [
+ *          [last week's page views for restaurant x1, last week's page views
+ * for restaurant x2, ...] [average page views for restaurant x1, average page
+ * views for restaurant x2, ...],
+ *     ],
+ *     'restaurantNames': [x1, x2, ...]
+ * }
+ */
+async function parseBarChartData() {
+  const restaurantPageViews = await getPageViewData();
+
+  const averages = getAveragePageViewArray(restaurantPageViews);
+  const lastWeekVals = getLastWeekPageViewArray();
+  const data = [lastWeekVals, averages];
+
+  const restaurantNames = getRestaurantNames(restaurantPageViews);
+
+  return {
+    'data': data,
+    'restaurantNames': restaurantNames,
+  };
+}
+
+/**
+ * Returns an array of the average page views for each restaurant.
+ * @param restaurantPageViews
+ * @return array of averages or empty array if no data
+ */
+function getAveragePageViewArray(restaurantPageViews) {
+  // Check for if there is no data stored yet
+  if (restaurantPageViews == null || restaurantPageViews.length <= 0) {
+    return [];
+  }
+
+  const averagePageViews = [];
+  // Iterate through all page view data for each restaurant
+  // and push average for each to return array
+  for (let i = 0; i < restaurantPageViews.length; i++) {
+    const currRestaurantPageViews = restaurantPageViews[i].pageViews;
+    let sum = 0;
+    for (let j = 0; j < currRestaurantPageViews.length; j++) {
+      sum += currRestaurantPageViews[j].count;
+    }
+    const count = currRestaurantPageViews.length;
+    const average = sum / count;
```

Comment:
is it possible for a restaurant to have no page views and cause this to div by zero?